From 4408dddb7af7a297ed6e0a758f45a4bfa23ea1b4 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 3 Feb 2019 10:58:27 +0200
Subject: [PATCH 0978/1051] dts: marvell: add CM3 SRAM memory to cp115 ethernet
 device tree

CM3 SRAM address space would be used for Flow Control configuration.

Change-Id: Ifdacf76e16e752ea4bc09448d6d2abcc57b75d85
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3419
Reviewed-by: Marcin Wojtas <Marcin.Wojtas@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 Documentation/devicetree/bindings/net/marvell-pp2.txt |  3 +++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi         | 10 ++++++++++
 2 files changed, 13 insertions(+)

diff --git a/Documentation/devicetree/bindings/net/marvell-pp2.txt b/Documentation/devicetree/bindings/net/marvell-pp2.txt
index fc019df0d863..ab7787b94f70 100644
--- a/Documentation/devicetree/bindings/net/marvell-pp2.txt
+++ b/Documentation/devicetree/bindings/net/marvell-pp2.txt
@@ -37,6 +37,7 @@ Required properties (port):
   GOP (Group Of Ports) point of view. This ID is used to index the
   per-port registers in the second register area.
 - phy-mode: See ethernet.txt file in the same directory
+- cm3-mem: phandle to CM3 SRAM definitions
 
 Optional properties (port):
 
@@ -60,6 +61,8 @@ ethernet@f0000 {
 	clocks = <&gateclk 3>, <&gateclk 19>;
 	clock-names = "pp_clk", "gop_clk";
 
+	cm3-mem = <&cm3_sram>;
+
 	eth0: eth0@c4000 {
 		interrupts = <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>;
 		port-id = <0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index d2e4387ea494..fecb50eef913 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -51,6 +51,8 @@
 			status = "disabled";
 			dma-coherent;
 
+			cm3-mem = <&CP110_LABEL(cm3_sram)>;
+
 			CP110_LABEL(eth0): eth0 {
 				interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>,
 					<ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>,
@@ -176,6 +178,14 @@
 			msi-parent = <&gicp>;
 		};
 
+		CP110_LABEL(cm3_sram): cm3@220000 {
+			compatible = "mmio-sram";
+			reg = <0x220000 0x800>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0 0x220000 0x800>;
+		};
+
 		CP110_LABEL(rtc): rtc@284000 {
 			compatible = "marvell,armada-8k-rtc";
 			reg = <0x284000 0x20>, <0x284080 0x24>;
-- 
2.17.1

